gcs/src/test/scala/org/broadinstitute/monster/storage/gcs/GcsApiSpec.scala

```diff
@@ -198,4 +219,203 @@ class GcsApiSpec extends FlatSpec with Matchers {
       }
       .unsafeRunSync()
   }
+
+  // statObject
+  it should "return true if a GCS object exists" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(Response[IO](body = Stream.emits("{}".getBytes())))
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (true -> None)
+  }
+
+  it should "return false if a GCS object does not exist" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(Response[IO](status = Status.NotFound))
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (false -> None)
+  }
+
+  it should "return the md5 of an existing object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.GET
+      req.uri shouldBe statObjectURI
+      Resource.pure(
+        Response[IO](
+          body = Stream.emits(s"""{"${GcsApi.ObjectMd5Key}": "$theMd5"}""".getBytes())
+        )
+      )
+    })
+
+    api
+      .statObject(bucket, path)
+      .unsafeRunSync() shouldBe (true -> Some(theMd5))
+  }
+
+  // createObject
+  it should "create a GCS object" in {
+    val api = new GcsApi(req => {
+      req.method shouldBe Method.POST
+      req.uri shouldBe createObjectURI
```

Comment:
Other things we should check:
1. The content-type of the request should be "multipart"
2. The body should parse as multipart, with two actual parts
3. The first part should have a content-type of JSON, and actually parse as JSON
4. The object name / md5 should be included in the JSON
5. The second part should have the content-type provided by the caller
6. The bytes of the second part should be the bytes provided by the caller

I can help walk you through adding these checks